### PR TITLE
fix: Cake Pool slider reset after ROI

### PR DESCRIPTION
**Pull request by ChefMomota**
https://user-images.githubusercontent.com/98292246/168784066-518878e9-e80a-48fc-a487-34d8a5c8a554.mov

### CODE REVIEW

File: src/views/Pools/components/LockedPool/Common/BalanceField.tsx [hunk 1]
```diff
@@ -36,6 +36,13 @@ const BalanceField: React.FC<PropsType> = ({
   const [percent, setPercent] = useState(0)
   const { userNotEnoughCake, notEnoughErrorMessage } = useUserEnoughCakeValidator(lockedAmount, stakingTokenBalance)
 
+  useEffect(() => {
+    const lockedAmountBigNumber = new BigNumber(lockedAmount)
+    if (lockedAmountBigNumber.gt(0)) {
+      handleStakeInputChange(lockedAmount)
+    }
+  }, [])
```
Comment (chefilip) [inline src/views/Pools/components/LockedPool/Common/BalanceField.tsx:44]: missing deps here?

**Comment by chef-jojo** [comment]
I wonder can we just treat the `lockedAmount` as single source, and compute the percentage on the fly.

File: src/views/Pools/components/LockedPool/Common/BalanceField.tsx (hunk 1 above)
Comment (ChefMomota) [reply src/views/Pools/components/LockedPool/Common/BalanceField.tsx]: Update watch the `lockedAmount`

File: src/views/Pools/components/LockedPool/Common/BalanceField.tsx [hunk 2]
```diff
@@ -36,19 +36,22 @@ const BalanceField: React.FC<PropsType> = ({
   const [percent, setPercent] = useState(0)
   const { userNotEnoughCake, notEnoughErrorMessage } = useUserEnoughCakeValidator(lockedAmount, stakingTokenBalance)
 
+  useEffect(() => {
+    const amount = new BigNumber(lockedAmount)
+    if (amount.gt(0)) {
+      const convertedInput = amount.multipliedBy(BIG_TEN.pow(stakingDecimals))
```
Comment (memoyil) [inline src/views/Pools/components/LockedPool/Common/BalanceField.tsx:42]: BIG_TEN.pow(stakingDecimals) can be extracted to a state by using initial state as function. Therefore we will not calculate it each render

Comment (memoyil) [reply src/views/Pools/components/LockedPool/Common/BalanceField.tsx:42]: Ignore my comment above, the correct approach should be a create a state with initalstate as function  and a useEffect that sets that state when stakingDecimals change.

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**